sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClientBuilder.java

```diff
@@ -100,98 +69,32 @@ public BlobClient buildBlobClient() {
-     */
-    public BlockBlobClient buildBlockBlobClient() {
-        return new BlockBlobClient(buildBlockBlobAsyncClient());
-    }
-
-    /**
-     * Creates a {@link BlockBlobAsyncClient} based on options set in the Builder. BlockBlobAsyncClients are used to
-     * perform generic upload operations such as {@link BlockBlobAsyncClient#uploadFromFile(String) upload from file}
-     * and block blob specific operations such as {@link BlockBlobAsyncClient#stageBlockWithResponse(String, Flux, long,
-     * LeaseAccessConditions) stage block} and {@link BlockBlobAsyncClient#commitBlockList(List) commit block list},
-     * only use this when the blob is known to be a block blob.
-     *
-     * @return a {@link BlockBlobAsyncClient} created from the configurations in this builder.
-     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
-     */
-    public BlockBlobAsyncClient buildBlockBlobAsyncClient() {
-        return new BlockBlobAsyncClient(constructImpl(), snapshot, cpk);
-    }
+        Objects.requireNonNull(containerName);
+        Objects.requireNonNull(blobName);
 
-    /**
-     * Creates a {@link PageBlobClient} based on options set in the Builder. PageBlobClients are used to perform page
-     * blob specific operations such as {@link PageBlobClient#uploadPages(PageRange, InputStream) upload pages} and
-     * {@link PageBlobClient#clearPages(PageRange) clear pages}, only use this when the blob is known to be a page
-     * blob.
-     *
-     * @return a {@link PageBlobClient} created from the configurations in this builder.
-     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
-     */
-    public PageBlobClient buildPageBlobClient() {
-        return new PageBlobClient(buildPageBlobAsyncClient());
-    }
+        HttpPipeline pipeline = super.getPipeline();
+        if (pipeline == null) {
+            pipeline = super.buildPipeline();
+        }
 
-    /**
-     * Creates a {@link PageBlobAsyncClient} based on options set in the Builder. PageBlobAsyncClients are used to
-     * perform page blob specific operations such as {@link PageBlobAsyncClient#uploadPages(PageRange, Flux) upload
-     * pages} and {@link PageBlobAsyncClient#clearPages(PageRange) clear pages}, only use this when the blob is known to
-     * be a page blob.
-     *
-     * @return a {@link PageBlobAsyncClient} created from the configurations in this builder.
-     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
-     */
-    public PageBlobAsyncClient buildPageBlobAsyncClient() {
-        return new PageBlobAsyncClient(constructImpl(), snapshot, cpk);
+        return new BlobAsyncClient(new AzureBlobStorageBuilder()
+            .url(String.format("%s/%s/%s", endpoint, containerName, blobName))
+            .pipeline(pipeline)
+            .build(), snapshot, cpk);
```

Comment:
What is `cpk`? This isn't a very intuitive variable name.

Reply:
cpk stands for customer provided key, and it belongs to a feature of the same name.

Reply:
We need reference to this feature name in some sense, as we need to easily differentiate it from client side encryption, the previous feature where customers managed their encryption keys, before the service was able to do that work for the customer.

Reply:
Changed field and getter to customerProvidedKey and getCustomerProvidedKey.